File: Number of Provinces/Number of Provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        n = len(isConnected)
        ctr = 0
        vis = [False] * n
        def dfs(node: int):
            vis[node] = True
            for i in range(n):
                if not vis[i] and isConnected[node][i]:
                    dfs(i)
        for i in range(n):
            if not vis[i]:
                ctr += 1
                dfs(i)
        return ctr

File: Number of Provinces/Number of Provinces.cpp
class Solution
{
public:
    void dfs(int node, int &n, vector<vector<int>> &g, vector<bool> &vis)
    {
        vis[node] = 1;
        for (int i = 0; i < n; i++)
        {
            if (!vis[i] && g[node][i])
                dfs(i, n, g, vis);
        }
    }
    int findCircleNum(vector<vector<int>> &isConnected)
    {
        int n = isConnected.size(), ctr = 0;
        vector<bool> vis(n, 0);
        for (int i = 0; i < n; i++)
        {
            if (!vis[i])
            {
                ctr++;
                dfs(i, n, isConnected, vis);
            }
        }
        return ctr;
    }
};